main.go

```diff
@@ -123,22 +86,45 @@ func runTemplate(c *cli.Context) error {
 
 	g := NewGomplate(data, lDelim, rDelim)
 
-	inputs := readInputs(c.String("in"), c.StringSlice("file"))
+	if err := validateInOutOptions(c); err != nil {
+		return err
+	}
 
-	outputs := c.StringSlice("out")
-	if len(outputs) == 0 {
-		outputs = []string{"-"}
+	if c.String("input-dir") != "" {
+		return processInputDir(c, g)
 	}
 
-	for n, input := range inputs {
-		out, err := openOutFile(outputs[n])
-		if err != nil {
-			return err
-		}
-		defer out.Close() // nolint: errcheck
-		g.RunTemplate(input, out)
+	return processInputFiles(c, g)
+}
+
+// Called from process ...
+func renderTemplate(g *Gomplate, inString string, outPath string) error {
+	outFile, err := openOutFile(outPath)
+	if err != nil {
+		return err
 	}
+	defer checkClose(outFile, &err)
+	g.RunTemplate(inString, outFile)
+	return nil
+}
 
+func validateInOutOptions(c *cli.Context) error {
```

Comment:
While this is totally necessary, I really don't like that this has to exist... I've been thinking of migrating to `github.com/jawher/mow.cli` for a while and maybe it's time to do that. It can handle option exclusion pretty well, so this validation would be unnecessary. Maybe I'll work on migrating to `mow.cli` in the next little while.